### PR TITLE
option: adds debugfailed option

### PR DESCRIPTION
**Pull request by catenacyber**
Prints contents of files in output directory of failed tests
Intent is to debug flasky behaviors

cf https://redmine.openinfosecfoundation.org/issues/3096

We should then update .travis.yml in suricata repository

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -883,6 +888,18 @@ def main():
...
+            print("")
+            print("Failed tests debug output:")
+        for dirpath in failedLogs:
+            print("- Test %s:" % os.path.basename(dirpath))
+            for r, d, f in os.walk(dirpath+"/output"):
+                for fname in f:
+                    print("  - %s" % fname)
+                    fcontents = open(dirpath+"/output/"+fname, "r")
+                    print(fcontents.read())
+
```
Comment (inashivb) [inline run.py]: So, just to be sure, we want to dump the information about why a test failed after running all the tests and not right when they fail?

File: run.py [hunk 2]
```diff
@@ -883,6 +888,18 @@ def main():
...
+    if args.debugfailed:
+        if len(failedLogs) > 0:
+            print("")
+            print("Failed tests debug output:")
+        for dirpath in failedLogs:
+            print("- Test %s:" % os.path.basename(dirpath))
+            for r, d, f in os.walk(dirpath+"/output"):
+                for fname in f:
+                    print("  - %s" % fname)
+                    fcontents = open(dirpath+"/output/"+fname, "r")
```
Comment (inashivb) [inline run.py]: This file pointer would have to be closed. Maybe a context manager with a [`with` statement](https://www.python.org/dev/peps/pep-0343/) would be better.
```
with open(filename) as fp:
    # do something with file
# unindent shall close the file now
```

Comment (inashivb) [reply run.py]: nit: spaces around operators

File: run.py (hunk 1 above)
Comment (catenacyber) [reply run.py]: Hmm... we could do it right away
This is a design choice about how a human would better read it

File: run.py (hunk 2 above)
Comment (catenacyber) [reply run.py]: ok for both

**Pull request closed without merging**